cmd/core/main.go

```diff
@@ -151,3 +163,45 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// setupChecks is used to create readiness&liveness probes
+func setupChecks(mgr ctrl.Manager) {
+	setupLog.Info("creating readiness/health check")
+	if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to create ready check")
+		os.Exit(1)
+	}
+
+	if err := mgr.AddHealthzCheck("ping", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to create health check")
+		os.Exit(1)
+	}
+}
+
+// waitSecretVolume waits for webhook secret ready to avoid mgr running crash
+func waitSecretVolume(certDir string) {
+	timeout := 300 * time.Second
+	interval := 2 * time.Second
+
+	start := time.Now()
+	for {
+		time.Sleep(interval)
+		if time.Since(start) > timeout {
+			setupLog.Error(errors.New("cannot get webhook secret "), "get webhook secret timeout")
+			os.Exit(1)
+		}
+		setupLog.Info("waiting admission webhook secret ... ")
```

Comment:
This log is important, we should add some more information such as 'xxx times have checked', 'xxx seconds left to timeout'...

Reply:
got it.